### PR TITLE
Fixes IV Drip

### LINKED ISSUE #939: IVs don't give non-blood
As in, they seemingly won't give other chems if they're placed in an IV drop and set to injecting.

## Reproduction:

1) Insert beaker/bottle of a selected chemical into an IV drip.
2) Ensure IV is on inject mode.
3) Connect yourself, and analyze yourself with a HF2.
4) Observe... nothing?

### PR DESCRIPTION
**Pull request by psykzz**
## About The Pull Request

Fixes #939

### CODE REVIEW

File: code/game/machinery/iv_drip.dm [hunk 1]
```diff
@@ -87,53 +87,56 @@
...
-				return
+		if(!istype(T)) return
+		if(!T.dna)
+			return
+		if(NOCLONE in T.mutations)
+			return
 
-			if(T.species && T.species.species_flags & NO_BLOOD)
-				return
+		if(T?.species.species_flags & NO_BLOOD)
```
Comment (Menshin) [inline code/game/machinery/iv_drip.dm]:
```suggestion
		if(T.species?.species_flags & NO_BLOOD)
```
We checked for `T` in the `istype(T)` a few lines above.

File: code/game/machinery/iv_drip.dm [hunk 2]
```diff
@@ -87,53 +87,56 @@
...
+		return
+
+	if(!beaker)
+		return
+
+	// Give blood
+	if(mode)
+		if(beaker.volume > 0)
+			var/transfer_amount = REAGENTS_METABOLISM
+			if(istype(src.beaker, /obj/item/reagent_container/blood))
```
Comment (Menshin) [inline code/game/machinery/iv_drip.dm]:
```suggestion
			if(istype(beaker, /obj/item/reagent_container/blood))
```

File: code/game/machinery/iv_drip.dm [hunk 3]
```diff
@@ -87,53 +87,56 @@
...
-			if(amount == 0)
-				if(prob(5)) visible_message("\The [src] pings.")
-				return
+	// Take blood
+	else
+		var/amount = beaker.reagents.maximum_volume - beaker.reagents.total_volume
+		amount = min(amount, 4)
+		// If the beaker is full, ping
+		if(amount == 0)
+			if(prob(5)) visible_message("\The [src] pings.")
```
Comment (Menshin) [inline code/game/machinery/iv_drip.dm]: I think same line control statements is frowned upon here.

File: code/game/machinery/iv_drip.dm [hunk 4]
```diff
@@ -87,53 +87,56 @@
...
 
-			var/mob/living/carbon/human/T = attached
+		var/mob/living/carbon/human/T = attached
 
-			if(!istype(T)) return
-			if(!T.dna)
-				return
-			if(NOCLONE in T.mutations)
-				return
+		if(!istype(T)) return
```
Comment (Menshin) [inline code/game/machinery/iv_drip.dm]: Same here.

File: code/game/machinery/iv_drip.dm [hunk 5]
```diff
@@ -87,53 +87,56 @@
...
-			if(T.species && T.species.species_flags & NO_BLOOD)
-				return
+		if(T?.species.species_flags & NO_BLOOD)
+			return
 
-			// If the human is losing too much blood, beep.
-			if(T.blood_volume < BLOOD_VOLUME_SAFE) if(prob(5))
-				visible_message("\The [src] beeps loudly.")
+		// If the human is losing too much blood, beep.
+		if(T.blood_volume < BLOOD_VOLUME_SAFE) if(prob(5))
```
Comment (Menshin) [inline code/game/machinery/iv_drip.dm]: Two `if` on the same line, probably not intended.

File: code/game/machinery/iv_drip.dm (hunk 1 above)
Comment (psykzz) [reply code/game/machinery/iv_drip.dm]: You're right this was misplaced, thanks.

**Comment by Menshin** [comment]
👍

**Comment by Rohesie** [comment]
Damn, merged this early by mistake. Oh well.

**Pull request merged**